pyccel/ast/internals.py

```diff
@@ -182,6 +182,21 @@ class PyccelSymbol(str, Immutable):
     >>> x = PyccelSymbol('x')
     x
     """
+    _is_temp = False
+
+    def __new__(cls, name, is_temp=False):
+        return super().__new__(cls, name)
+
+    def __init__(self, name, is_temp=False):
+        self._is_temp = is_temp
+        super().__init__()
```

Comment:
Why do we need a class attribute `_is_temp` if we assign the instance attribute `_is_temp`  (which effectively overwrites the class attribute) in the `__init__`?

Reply:
because PyccelSymbol inherits from `str`  which has no argument called '_is_temp`,` __new__() got an unexpected keyword argument 'is_temp'`  will be raised, that's why we need to redefine `__new__` in PyccelSymbol, so that we can use `is_temp` in PyccelSymbol.

Reply:
because PyccelSymbol inherits from str which has no argument called '_is_temp, `new() got an unexpected keyword argument 'is_temp'` will be raised, that's why we need to redefine `__new__` in PyccelSymbol, so that we can use `is_temp` in PyccelSymbol.

Reply:
A class attribute is an attribute which can be called like: `PyccelSymbol._is_temp` without an instance of the class. This is what you have defined at line 185

Reply:
An instance attribute is an attribute which is attached to `self`, i.e. the one defined at line 191

Reply:
@mostafaMamoni I am specifically referring to lines 185 and 191, as explained in @EmilyBourne's comments.

I think that you can just remove line 185.